test/unit/test.py

```diff
@@ -288,20 +288,21 @@ def test_spec_or_default_1(self):
 
     def test_spec_or_default_2(self):
         test_spec = {
-            "existing_key": "true",
+            "existing_key": True,  # Changed from "true" to True
             "existing_key_2": True,
-            "existing_key_3": "false",
+            "existing_key_3": False,  # Changed from "false" to False
```

Comment:
You can remove a lot of these tests (including some of the checks in line 308 and following) because we expect the the types to be preserved